pynsee/utils/save_df.py

```diff
@@ -0,0 +1,126 @@
+    if (mdate is not None) and (day_lapse is not None):
+        strg_print += f"Creation date: {mdate:%Y-%m-%d}"
+        
+        if day_lapse >= 2:
+            strg_print += f", {day_lapse} days ago\n"
+        if day_lapse == 1:
+            strg_print += ", yesterday\n"
+        if day_lapse == 0:
+            strg_print += ", today\n"
+            
+    strg_print += "Set update=True to get the most up-to-date data"
+
+    logger.info(strg_print)
+
+def save_df(obj=pd.DataFrame, parquet=True, day_lapse_max=None):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            
+            data_folder = _create_insee_folder()
+            string_file_arg = [str(kwargs[a]) for a in kwargs.keys() if a not in ['update', 'silent']]
+            string_file_arg += [func.__name__] + [str(a) for a in args]            
+            file_name = data_folder + "/" + _hash(''.join(string_file_arg)) 
+            
+            if parquet:
+                file_name += ".parquet"
+            else:
+                file_name += ".pkl"            
+                                    
+            update = kwargs.get("update", False)
+
+            silent = kwargs.get("silent", False)
+            
+            if os.path.exists(file_name):
+                file_date_last_modif = datetime.datetime.fromtimestamp(
+                    os.path.getmtime(file_name)
+                )
+
+                insee_date_time_now = kwargs.get("insee_date_test", False)
+                
+                if insee_date_time_now is not False:                    
+                                        
+                    insee_date_time_now = datetime.datetime.strptime(
+                        insee_date_time_now, "%Y-%m-%d %H:%M:%S.%f"
+                    )
+                else:
+                    insee_date_time_now = datetime.datetime.now()
+
+                day_lapse = (insee_date_time_now - file_date_last_modif).days
+
+                if day_lapse_max is not None:
+                    if day_lapse > day_lapse_max:
+                        update = True                    
+            
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", category=ShapelyDeprecationWarning)
+
+                if (not os.path.exists(file_name)) or update:   
+
+                    df = func(*args, **kwargs)                    
+                    try:
+                        if parquet:
+                            df.to_parquet(file_name)
+                        else:
+                            df.to_csv(file_name, index=False)
+                            
+                    except Exception as e:
+                        warnings.warn(str(e))
+                        warnings.warn(f'Error, file not saved:\n{file_name}\n{df}')
+                        warnings.warn('\n')
+                    
+                    df = obj(df) 
```

Comment:
remove and wrap the return instead

Reply:
4cf3d511b1bd7942b9075ccc463d4664fdd5ff5c should tackle the raised issue